SoftFox/SoftFox/SoftFox.cpp

```diff
@@ -112,12 +112,47 @@ void SoftFox::run()
 			playerX -= PLAYER_MOVEMENT_SPEED;
 		if (keyboardState[SDL_SCANCODE_RIGHT])
 			playerX += PLAYER_MOVEMENT_SPEED;
+		
+
+		SDL_Rect playerBox = { playerX, playerY, tileSize / 2, tileSize };
+
+		playerY += gravity;
+		
 		/*
-		if (physics->isCollision(playerSprite, platformSprite))
+		int platformX = tileSize * level->getWallX() + tileSize / 2;// + tileSize;
+		int platformY = tileSize * level->getWallY() + tileSize / 2;// - tileSize/4;
+		SDL_Rect platformBox = { platformX, platformY, tileSize, tileSize };
+
+		if (physics->isCollision(playerBox, platformBox))
 		{
-			PLAYER_MOVEMENT_SPEED = 0;
+			playerY -= upForce;
 		}
 		*/
+		
+		for (int y = 0; y < level->getHeight(); y++)
+		{
+			for (int x = 0; x < level->getWidth(); x++)
+			{
+				if (level->isWall(x, y))
+				{
+					int platformX[] = { tileSize * level->getWallX() + tileSize / 2 };
+					int platformY[] = { tileSize * level->getWallY() + tileSize / 2 };
+					for (int a = 0; a < size_t (platformX); a++)
+					{
+						int platformXA = platformX[a]*tileSize;
+						for (int b = 0; b < size_t (platformY); b++)
+						{
+							int platformYB = platformY[b] * tileSize;
+							SDL_Rect platformBox = { platformXA, platformYB, tileSize, tileSize };
```

Comment:
I **think** what you are trying to do in this function is something like this:
```
for each (x,y) in the level
  if there is a wall at (x,y)
    let platformBox be the rectangle for the wall at (x,y)
    if playerBox collides with platformBox
      subtract upForce from playerY
    end if
  end if
end for
```
In which case you don't need the loops over `a` and `b` -- inside the `if (level->isWall(x, y))` block you just need to calculate the pixel coordinates for the rectangle and then do the collision check. I think you have already tried to do that calculation with `platformX` and `platformY`, so it should work once you straighten out the syntax and get rid of the unneeded loops.